### PR TITLE
#309 (create `minos.common.DataTransferObject` class)

### PR DESCRIPTION
**Pull request by vladyslav-fenchak**

### CODE REVIEW

File: minos/common/model/abc/dto_model.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+from .model import (
+    MinosModel,
+)
+
+
+class DtoModel(MinosModel):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def from_avro_bytes(self, raw: bytes, **kwargs) -> t.Type[DtoModel]:
```
Comment (garciparedes) [inline minos/common/model/abc/dto_model.py:29]: This method must be a `classmethod`

File: minos/common/model/abc/dto_model.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    ModelField,
+)
+from .model import (
+    MinosModel,
+)
+
+
+class DtoModel(MinosModel):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```
Comment (garciparedes) [inline minos/common/model/abc/dto_model.py:27]: It's better not to define the `init` class if not any new attributes has been set.

File: tests/test_common/test_model/test_abc/test_dto_model.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+"""
+import unittest
+
+from minos.common import (
+    DtoModel,
+    MinosAvroProtocol,
+)
+
+
+class TestDtoField(unittest.IsolatedAsyncioTestCase):
```
Comment (garciparedes) [inline tests/test_common/test_model/test_abc/test_dto_model.py:16]: The class name of the test case does not matches the `"Test${CLASS_NAME}"` pattern

File: tests/test_common/test_model/test_abc/test_dto_model.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                "type": "record",
+            },
+        ]
+        serialized = MinosAvroProtocol.encode(data, schema)
+        self.assertEqual(True, isinstance(serialized, bytes))
+
+        dto_model = DtoModel()
+        dto_model.from_avro_bytes(serialized)
+
+        self.assertEqual(data["cost"], dto_model.fields["cost"].value)
```
Comment (garciparedes) [inline tests/test_common/test_model/test_abc/test_dto_model.py:60]: You can directly access the fields simply with `dto_model.cost` and `dto_model.username`

File: minos/common/model/abc/dto_model.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    def from_avro_bytes(self, raw: bytes, **kwargs) -> t.Type[DtoModel]:
+        """Build a single instance or a sequence of instances from bytes
+
+        :param raw: A bytes data.
+        :return: A single instance or a sequence of instances.
+        """
+
+        schema = MinosAvroProtocol().decode_schema(raw)
+        decoded = MinosAvroProtocol().decode(raw)
```
Comment (garciparedes) [inline minos/common/model/abc/dto_model.py:37]: This methods are static ones so you don't need the `()` at the end of `MinosAvroProtocol`

File: minos/common/model/abc/dto_model.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+)
+from .fields import (
+    ModelField,
+)
+from .model import (
+    MinosModel,
+)
+
+
+class DtoModel(MinosModel):
```
Comment (garciparedes) [inline minos/common/model/abc/dto_model.py:25]: I was thinking about a "fancy" name for this class and I think that one alternative could be to name it as `DataTransferModel`. What do you think about that? 🤔

**Review by garciparedes — CHANGES_REQUESTED** [review]
Good changes @vladyslav-fenchak, I think that will result in an awesome functionality of the `minos` framework! 🎉 

Only one more change request. Could you move the `dto_model.py` one level up (to be on the `models/` directory and also rename it simply as `dto.py`  🙂

File: minos/common/model/abc/dto_model.py (hunk 6 above)
Comment (vladyslav-fenchak) [reply minos/common/model/abc/dto_model.py:25]: Yes, I think it's better to call it that.

File: minos/common/model/abc/dto_model.py (hunk 5 above)
Comment (vladyslav-fenchak) [reply minos/common/model/abc/dto_model.py:37]: Right, done !!!

File: tests/test_common/test_model/test_abc/test_dto_model.py (hunk 4 above)
Comment (vladyslav-fenchak) [reply tests/test_common/test_model/test_abc/test_dto_model.py:60]: Right, done !!!

File: tests/test_common/test_model/test_abc/test_dto_model.py (hunk 3 above)
Comment (vladyslav-fenchak) [reply tests/test_common/test_model/test_abc/test_dto_model.py:16]: Right, done !!!

File: minos/common/model/abc/dto_model.py (hunk 2 above)
Comment (vladyslav-fenchak) [reply minos/common/model/abc/dto_model.py:27]: Right, done !!!

File: minos/common/model/abc/dto_model.py (hunk 1 above)
Comment (vladyslav-fenchak) [reply minos/common/model/abc/dto_model.py:29]: Ok, Done!!

**Review by andrea-mucci — COMMENTED** [review]
@vladyslav-fenchak please add more tests with the following data types:

1) list
2) array

File: minos/common/model/dto.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+from minos.common.model.declarative.abc import (
+    MinosModel,
+    ModelField,
+)
+from minos.common.protocol import (
+    MinosAvroProtocol,
+)
+
+
+class DataTransferObject(MinosModel):
```
Comment (garciparedes) [inline minos/common/model/dto.py:23]: Could you add a small description as a `docstring` @vladyslav-fenchak?

Comment (garciparedes) [inline minos/common/model/dto.py:23]: Also, you must inherit the `DynamicModel` class, that already provides the `from_avro_bytes` method that you need.

**Review by garciparedes — APPROVED** [review]
Hi @vladyslav-fenchak, I really like this new feature! 🚀 

Resolve the inheritance issues that I told you and I'll proceed to approve this Pull Request. 🎉

**Review by garciparedes — CHANGES_REQUESTED** [review]
Hi @vladyslav-fenchak, I really like this new feature! 🚀 

Resolve the inheritance issues that I told you and I'll proceed to approve this Pull Request. 🎉

File: minos/common/model/dto.py (hunk 7 above)
Comment (vladyslav-fenchak) [reply minos/common/model/dto.py]: Done!

Comment (vladyslav-fenchak) [reply minos/common/model/dto.py]: Done!

File: minos/common/model/dto.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+from minos.common.model.declarative.abc import (
+    ModelField,
+)
+from minos.common.model.dynamic.abc import (
+    DynamicModel,
+)
+from minos.common.protocol import (
+    MinosAvroProtocol,
+)
```
Comment (garciparedes) [inline minos/common/model/dto.py:22]: Relative imports are a better alternative.

File: minos/common/model/dto.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        self.build_field(schema, decoded)
+
+    def build_field(self, schema: dict, value: t.Any) -> t.NoReturn:
+        """Build Model Fields from avro schema and value
+
+        :param schema: Part of avro schema
+        :param value: Decoded avro values
+        """
+        field_name = schema["name"]
+        self._fields[field_name] = ModelField.from_avro(schema, value)
```
Comment (garciparedes) [inline minos/common/model/dto.py:59]: This code is already implemented in the `DynamicModel` base class.

**Review by garciparedes — CHANGES_REQUESTED** [review]

**Review by andrea-mucci — APPROVED** [review]

**Review by garciparedes — APPROVED** [review]

**Review by garciparedes — APPROVED** [review]

**Review by andrea-mucci — APPROVED** [review]

**Pull request merged**